### PR TITLE
Enables programs to bring their own logger

### LINKED ISSUE #17: Pluggable logging
It would be nice if programs that integrate Smudge could bring their own logger.

I'm willing to create a PR for this. Hopefully somewhere in the coming weeks.

### PR DESCRIPTION
**Pull request by eelcocramer**
Closes #17

### CODE REVIEW

**Comment by eelcocramer** [comment]
Please note that the changes of PR #16 are also part of this PR!

**Comment by clockworksoul** [comment]
Thanks for these! Apologies for not being able to review swiftly... work has been a little demanding lately. I'll have time this Saturday, though!

**Comment by clockworksoul** [comment]
I'm generally in favor of adding this, but it does add the dependency on `github.com/rs/zerolog/log` and all of the problems that come with it. For that reason I've consciously avoided adding any dependencies at all (I have heard some encouraging news regarding Golang package management recently, however!)

That being said, If Smudge is going to see wider adoption I'll need to be a little more flexible, but I still want to keep external dependencies to a minimum.

File: membership.go [hunk 1]
```diff
@@ -192,13 +193,18 @@ func PingNode(node *Node) error {
 // Byte  0      - 1 byte character byte length N
 // Bytes 1 to N - Cluster name bytes
 // Bytes N+1... - A message (without members)
-func decodeMulticastAnnounceBytes(bytes []byte) (string, []byte) {
+func decodeMulticastAnnounceBytes(bytes []byte) (string, []byte, error) {
```
Comment (clockworksoul) [inline membership.go:196]: Now that this change is in master, would you mind rebasing and re-pushing?

File: smudge/smudge.go [hunk 2]
```diff
@@ -20,9 +20,47 @@ import (
 	"flag"
 	"fmt"
 	"github.com/clockworksoul/smudge"
-	"log"
+	"github.com/rs/zerolog/log"
```
Comment (clockworksoul) [inline smudge/smudge.go:23]: Adding this dependency breaks `make image` and the build instructions in the README. Would you mind adding the `go get` to the Dockerfile and update the README appropriately?

**Review by clockworksoul — CHANGES_REQUESTED** [review]
Minor changes requested due to dependency introduction.

File: smudge/smudge.go (hunk 2 above)
Comment (eelcocramer) [reply smudge/smudge.go:23]: You are right. I was not intended to commit `smudge.go`. It was my PoC test for adding an external logger. I will remove it from the PR.

**Pull request closed without merging**